compiler/src/iree/compiler/Dialect/VM/Conversion/VMToEmitC/ConvertVMToEmitC.cpp

```diff
@@ -4836,8 +4834,39 @@ class ConvertVMToEmitCPass
     }
 
     SmallVector<std::string> importShims;
+
+    // The conversion of `call/call.variadic` ops on imported functions expects
+    // import ops to be rewritten to compiler generated shim functions. To
+    // ensure this we only rewrite `import` ops first.
+    auto rewriteImports = [&module, &importShims](MLIRContext &ctx) {
```

Comment:
This is a lot of work to replace a single op - instead of using the conversion infrastructure here you could just walk the ops immediately under the module and do what the body of the ImportOpConversion does by making it a function call - then it's a traversal of a few dozen ops and some simple in-place replacements. The conversion infrastructure will recursively walk all ops in the program, do a bunch of additional verification, pass through multiple times, and be harder to debug (especially since you don't actually replace the import op and are relying on the dynamic legality check, which is also tricky).

So:
```c++
for (auto importOp : module.getOps<IREE:VM::ImportOp>()) {
  createImportShim(importOp);
}
...
void createImportShim(IREE::VM::ImportOp importOp) {
  OpBuilder builder(importOp);  // rewriter in the existing code
  << contents of ImportOpConversion >>
}
```